src/lib/Characteristic.ts

```diff
@@ -1896,9 +1899,10 @@ export class Characteristic extends EventEmitter {
    * in the future and throws an error which can't be converted to a valid value.
    *
    * @param value - The value received from the API call
+   * @param characteristicWarningType - Type of the warning message
```

Comment:
I would propose to be a bit more specific.

```suggestion
   * @param characteristicWarningType - Optionally defines the warning type to use when raising a {@link CharacteristicEventTypes.CHARACTERISTIC_WARNING}.
```